lib/Frontend/CompilerInvocation.cpp

```diff
@@ -1323,6 +1323,11 @@ static bool ParseLangArgs(LangOptions &Opts, ArgList &Args,
       Diags.diagnose(SourceLoc(), diag::evolution_with_embedded);
       HadError = true;
     }
+
+    if (FrontendOpts.InputsAndOutputs.hasPrimaryInputs()) {
```

Comment:
I don’t really understand this. Primaries and WMO don’t really mix. Are you trying to detect the optimization mode, or the compilation mode?

Reply:
Thanks for chiming in here, I could use some help with this :)

What I’m trying to do is to require WMO, the compilation mode. I thought that the presense of a primary file is a clear sign that we’re *not* doing WMO (and vice versa). Is that not true?